src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -52,89 +52,55 @@ case class AddFileMetadataToDeposit(row: Int, entry: (DatasetID, Dataset))(impli
 object AddFileMetadataToDeposit {
 
   def writeFileMetadataXml(row: Int, datasetID: DatasetID)(implicit settings: Settings): Try[Unit] = {
-    Try {
-      outputFileMetadataFile(datasetID).writeXml(datasetToFileXml(datasetID))
-    } recoverWith {
-      case NonFatal(e) => Failure(ActionException(row, s"Could not write file meta data: $e", e))
-    }
+    datasetToFileXml(datasetID)
+      .map(outputFileMetadataFile(datasetID).writeXml(_))
```

Comment:
the `writeXml` is inside a `map` on the `Try` class. The methods on this class are all fail safe using try-catch blocks. So if `writeXml` fails, the exception that is thrown will be caught by the surrounding `map` and cause the result to be a `Failure`.